adapter/datadog/logger.go

```diff
@@ -0,0 +1,64 @@
+package datadog
+
+import (
+	"errors"
+	"strings"
+
+	coopLogger "github.com/coopnorge/go-logger"
+)
+
+// Logger is a logging adapter between gopkg.in/DataDog/dd-trace-go.v1/ddtrace
+// an go-logger, do not create this directly, use NewLogger()
+type Logger struct {
+	instance *coopLogger.Logger
+}
+
+// NewLogger creates a new Gorm logger that passes message to go-logger
+//
+// To inject the logger into ddtrace use
+//
+//	package main
+//
+//	import (
+//		datadogLogger "github.com/coopnorge/go-logger/adapter/datadog"
+//
+//		"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+//	)
+//
+//	func main() {
+//		l, err =: gormLogger.NewLogger(gormLogger.WithGlobalLogger())
+//		ddtrace.UseLogger(l)
+//	}
+func NewLogger(opts ...LoggerOption) (*Logger, error) {
+	logger := &Logger{}
+	for _, opt := range opts {
+		opt.Apply(logger)
+	}
+	if logger.instance == nil {
+		return nil, errors.New("No go-logger instance provided, use WithGlobalLogger() or WithLogger() to configure the logger")
+	}
+	return logger, nil
+}
+
+// Log writes statements to the log
+func (l *Logger) Log(msg string) {
+	// Logs from gopkg.in/DataDog/dd-trace-go.v1/ddtrace will contain keywords
+	// specifying the level of the log.
+	if strings.Contains(msg, "DEBUG") {
+		l.instance.Debug(msg)
+		return
+	}
+	if strings.Contains(msg, "INFO") {
+		l.instance.Info(msg)
+		return
+	}
+	if strings.Contains(msg, "WARN") {
+		l.instance.Warn(msg)
+		return
+	}
+	if strings.Contains(msg, "ERROR") {
+		l.instance.Error(msg)
+		return
+	}
+	l.instance.Info(msg)
+}
```

Comment:
I suggest we flip the order, so that we check in the order of highest severity to lowests, in case a message contains e.g. both `DEBUG` and `ERROR`, it would probably be best to log the message as Error-level.

Also maybe add an extra field to the fallback, stating that we could not determine the Datadog-level, and increase the level to Warn instead of Info?

Reply:
Fixed